tests/main/views/test_direct_award.py

```diff
@@ -1049,13 +1049,18 @@ def test_invalid_data_raises_400_and_has_validation_messages_but_remains_in_form
         assert res.status_code == 400
 
         doc = html.fromstring(res.get_data(as_text=True))
-        assert doc.xpath('count(//span[@class="validation-message"])') == 1
+        assert (doc.xpath('count(//span[@class="validation-message"])') == 1 or
+                doc.xpath('count(//span[@class="govuk-error-message"])') == 1)
         errors = doc.cssselect('div.govuk-error-summary a')
         assert len(errors) == 1
         assert len(errors[0].text_content()) > 0
 
         for field, value in data.items():
-            assert doc.xpath(f'//input[@name="{field}"]/@value')[0] == value
+            input_element = doc.xpath(f'//input[@name="{field}"]')[0]
+            if (value is None or value == ''):
+                assert(not input_element.xpath('/@value'))
```

Comment:
I am a bit confused what are we asserting in this case?

Reply:
You're right, this is confusing and hacky.

Previously the element being checked _always_ had a `value` attribute, even if the form input was empty.

With the GOVUK Frontend component, if the input is empty, it won't have a `value` attribute, so this is essentially checking that there's no element with a `value` attribute.

I've pushed a change to make it clearer.

It's hacky - previously the element being checked always had a `value` attribute, but now it's possible to not have that attribute if the form value is empty.

So in the case where the input is empty, this is asserting that the element does not have a `value` attribute.

I'm not v. familiar with xpath, so I dunno if there's a way